### PR TITLE
change GPT2ForSequenceClassification inference accuracy tolerance

### LINKED ISSUE #123503: [inductor][cpu]GPT2ForSequenceClassification AMP static/dynamic shape default/cpp wrapper single thread accuracy crash
### 🐛 Describe the bug

```
loading model: 0it [00:02, ?it/s]cpu  eval  GPT2ForSequenceClassification      

E0402 16:13:08.429653 134289029980032 torch/_dynamo/utils.py:1405] RMSE (res-fp64): 0.00568, (ref-fp64): 0.00100 and shape=torch.Size([1, 2]). res.dtype: torch.bfloat16, multiplier: 3.000000, tol: 0.004000
E0402 16:13:08.429873 134289029980032 torch/_dynamo/utils.py:1319] Accuracy failed for key name logits
fail_accuracy
```

### Versions

</table><p>SW info</p><table border="1" class="dataframe table">
  <thead>
    <tr style="text-align: right;">
      <th>name</th>
      <th>target_branch</th>
      <th>target_commit</th>
      <th>refer_branch</th>
      <th>refer_commit</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>torchbench</td>
      <td>main</td>
      <td>d6015d42</td>
      <td>main</td>
      <td>d6015d42</td>
    </tr>
    <tr>
      <td>torch</td>
      <td>main</td>
      <td>781e8d2201c1e2aaeccbbc7b7b13f9322b481bc9</td>
      <td>main</td>
      <td>f0d461beacded34abe196c72ec4bcdb55bf01793</td>
    </tr>
    <tr>
      <td>torchvision</td>
      <td>main</td>
      <td>0.19.0a0+2c4665f</td>
      <td>main</td>
      <td>0.19.0a0+a0c79b3</td>
    </tr>
    <tr>
      <td>torchtext</td>
      <td>main</td>
      <td>0.16.0a0+b0ebddc</td>
      <td>main</td>
      <td>0.16.0a0+b0ebddc</td>
    </tr>
    <tr>
      <td>torchaudio</td>
      <td>main</td>
      <td>2.2.0a0+ea437b3</td>
      <td>main</td>
      <td>2.2.0a0+17a7081</td>
    </tr>
    <tr>
      <td>torchdata</td>
      <td>main</td>
      <td>0.7.1a0+0790338</td>
      <td>main</td>
      <td>0.7.1a0+0790338</td>
    </tr>
    <tr>
      <td>dynamo_benchmarks</td>
      <td>main</td>
      <td>nightly</td>
      <td>main</td>
      <td>nightly</td>
    </tr>
  </tbody>
</table>

</table>

Repro:
[inductor_single_run.sh](https://github.com/chuanqi129/inductor-tools/blob/main/scripts/modelbench/inductor_single_run.sh)
bash inductor_single_run.sh single inference accuracy huggingface GPT2ForSequenceClassification amp first static/dynamic default/cpp
[huggingface-GPT2ForSequenceClassification-inference-amp-static-cpp-single-accuracy-crash_guilty_commit.log](https://github.com/pytorch/pytorch/files/14893899/huggingface-GPT2ForSequenceClassification-inference-amp-static-cpp-single-accuracy-crash_guilty_commit.log)
Suspected guilty commit: https://github.com/pytorch/pytorch/commit/5152945441b2a1387f6e9fc8fe1531bfeb6cabe4
cc @ezyang @msaroufim @bdhirsh @anijain2305 @chauhang @WeizhuoZhang-intel @chuanqi129

### PR DESCRIPTION
**Pull request by Valentine233**
Fixes https://github.com/pytorch/pytorch/issues/123503.

https://github.com/pytorch/pytorch/pull/121866 makes GPT2ForSequenceClassification hit the SDPA pattern 18 and then encounter the accuracy issue. The issue only happens with BF16 inference single thread. This PR tends to increase the model tolerance from 4e-3 to 5e-3 and make the check pass. Note that the issue is due to some small implementation diff. For example, the sdpa math backend scales q, k before matmul for stability; the flash attention backend has more diffs as a new algorithm.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### CODE REVIEW

**Comment by sanchitintel** [comment]
Same issue with pattern 19

**Comment by Valentine233** [comment]
> Same issue with pattern 19.
> 
> > Note that the issue is due to some small implementation diff.
> 
> Hi @Valentine233, do you already know the root-cause (and is it something that can't be fixed asap)? Thanks!

Hi @sanchitintel, I found other sdpa pattern UTs, e.g. pattern 1, also had accuracy issues with BF16. As the implementations are not exactly the same, it is reasonable to relax the tolerance bounds in this case. Note that the SDPA math backend also meets the accuracy issues with the pattern match.

**Review by jgong5 — APPROVED** [review]
Could you please elaborate on the "small implementation difference" in the PR description?

**Comment by Valentine233** [comment]
Hi @drisspg , could you help review the PR? Thanks!

**Comment by Valentine233** [comment]
Hi @kit1980 , could you help review the PR? Thanks!

**Review by jansel — APPROVED** [review]
